bark/world/objects/agent.hpp

```diff
@@ -170,6 +170,10 @@ class Agent : public Object {
     history_ = history;
   }
 
+  void SetCurrentState(const State& current_state) {
+    history_.back().first = current_state;
```

Comment:
should we not fill the action also with an default value?

Reply:
Yeah, thats right there is missing the case of the history being empty. But the function is only called after the constructor where the first entry of the history is always set

Reply:
Yeah, thats right there is missing the case of the history being empty. But the function is only called after the constructor where the first entry of the history is always set